src/NServiceBus.NHibernate.AcceptanceTests/When_using_synchronized_session_via_container.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Threading.Tasks;
+using NServiceBus.AcceptanceTesting;
+using NServiceBus.AcceptanceTests.EndpointTemplates;
+using NUnit.Framework;
+
+namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using global::NHibernate;
+
+    [TestFixture]
+    public class When_using_synchronized_session_via_container : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_inject_synchronized_session_into_handler()
```

Comment:
Can we test `OnSaveChanges`

Reply:
It is tested in the unit tests e.g. `NHibernateAmbientTransactionSynchronizedStorageSessionTests` and two others in the same folder.